packages-exp/auth-exp/src/platform_browser/recaptcha/recaptcha_loader.test.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+
+import { FirebaseError } from '@firebase/util';
+
+import { testAuth } from '../../../test/mock_auth';
+import { Auth } from '../../model/auth';
+import { AuthWindow } from '../auth_window';
+import * as jsHelpers from '../load_js';
+import {
+  _JSLOAD_CALLBACK,
+  MOCK_RECAPTCHA_LOADER,
+  ReCaptchaLoader,
+  ReCaptchaLoaderImpl
+} from './recaptcha_loader';
+import { MockReCaptcha } from './recaptcha_mock';
+
+const WINDOW: AuthWindow = window;
+
+use(chaiAsPromised);
+use(sinonChai);
+
+describe('platform-browser/recaptcha/recaptcha_loader', () => {
+  let auth: Auth;
+
+  beforeEach(async () => {
+    auth = await testAuth();
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    delete WINDOW.grecaptcha;
+  });
+
+  describe('MockLoader', () => {
+    it('returns a MockRecaptcha instance', async () => {
+      expect(await MOCK_RECAPTCHA_LOADER.load(auth)).to.be.instanceOf(
+        MockReCaptcha
+      );
+    });
+  });
+
+  describe('RealLoader', () => {
+    let triggerNetworkTimeout: () => void;
+    let jsLoader: { resolve: () => void; reject: () => void };
+    let loader: ReCaptchaLoader;
+    const networkTimeoutId = 123;
+
+    beforeEach(() => {
+      sinon.stub(window, 'setTimeout').callsFake(cb => {
```

Comment:
can you extract this thing to a test helper?

Reply:
Done